src/ray/object_manager/pull_manager.cc

```diff
@@ -0,0 +1,132 @@
+#include "ray/object_manager/pull_manager.h"
+
+namespace ray {
+
+PullManager::PullManager(
+    NodeID &self_node_id, const std::function<bool(const ObjectID &)> object_is_local,
```

Comment:
Consider passing in a const ref to the local objects hashmap here. Not a strong preference, but I think it's a bit nicer when reading the code.

Reply:
Hmm I don't feel strongly about it, but I think const ref is as narrow as a lambda here. It makes it clear that this class has read-only access.

Reply:
You have the ability to do other things like iterate over the map, etc. which is a much broader set of APIs than a single function.

Reply:
I don't feel strongly about this at all, but one minor advantage of `object_is_local` is that it provides proof of existence only, which means we don't have to mock out/know things about the `LocalObjectInfo`.